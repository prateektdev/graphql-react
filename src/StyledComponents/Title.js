import styled from "styled-components";

const Title = styled.h1`
  font-size: 1.5em;
  display:block;
  text-align: center;
  color: #0d6efd;
`;
export default Title;