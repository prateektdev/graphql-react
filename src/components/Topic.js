import React, { useEffect, useState } from "react";
import Button from "../StyledComponents/Button";
import Input from "../StyledComponents/Input";
import Title from "../StyledComponents/Title";
import TopicResult from "./TopicResult";
//This is the main component for topics
const Topic = (props) => { 
  const [searchTerm, setSearchTerm] = useState(props.searchValue);  

  //using this hook to update the searchbar value if clicked from related topics
  useEffect(() => {
    setSearchTerm(props.searchValue);
  },[props.searchValue]); //update only if the related topic is clicked

  return (
    <React.Fragment>
      <nav className="navbar navbar-dark bg-dark">
        <div className="navbar-brand "><Title>Test task GraphQL</Title></div>
        <div className="form-inline">
          <Input
            value={searchTerm}
            onChange={(e) => setSearchTerm(e.target.value)}
            className="form-control" type="search" placeholder="Search" aria-label="Search" />
          <Button className="btn btn-outline-primary my-2 my-sm-0" type="submit"
            onClick={() => props.onSearchInputChange(searchTerm)}>Search</Button>
        </div>
      </nav>
       {/* The current Topic searched*/}
      <h1><span className="text-info text-uppercase">{props.searchValue}</span></h1>
      <TopicResult key={props.searchValue} value={props.searchValue} onChange={props.onSearchInputChange} ></TopicResult>
    </React.Fragment>
  )
}
export default Topic;