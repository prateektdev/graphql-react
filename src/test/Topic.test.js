import { MockedProvider } from "@apollo/client/testing";
import { mount, shallow } from "enzyme";
import Title from "../StyledComponents/Title";
import Topic from "../components/Topic";

describe("Topic ", () => {
  it("should render Topic component", () => {
    const wrapper = shallow(<MockedProvider   addTypename={false}><Topic /></MockedProvider>);
    expect(wrapper).toMatchSnapshot();
  });
  it('contains Title', () => {
    const wrapper = mount(<MockedProvider  addTypename={false}><Topic /></MockedProvider>);
    const value = wrapper.find(Title).text()
    expect(value).toEqual('Test task GraphQL')
  })
  it('contains props', () => {
    const wrapper = mount(<MockedProvider  addTypename={false}><Topic searchValue="react" /></MockedProvider>);  
    expect(wrapper.find(Topic).props().searchValue).toEqual('react')
  })
});