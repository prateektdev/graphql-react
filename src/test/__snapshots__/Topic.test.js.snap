// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic  should render Topic component 1`] = `ShallowWrapper {}`;
